cardano-cli/src/Cardano/CLI/Compatible/Exception.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Cardano.CLI.Compatible.Exception
+  ( CustomCliException (..)
+  , throwCliError
+  , fromEitherCli
+  , fromEitherIOCli
+  )
+where
+
+import           Cardano.Api
+
+import           GHC.Stack
+
+import           RIO
+
+-- | Custom exception type for CLI commands. Any custom errors created
+-- in `cardano-cl` should be wrapped in this exception type.
+data CustomCliException where
+  CustomCliException
+    :: (Show error, Typeable error, Error error)
+    => error -> CallStack -> CustomCliException
```

Comment:
Why explicit `CallStack` argument? The way it's used it can be an implicit one.

Reply:
Addressed this above.